modules/renter/download.go

```diff
@@ -11,114 +11,169 @@ import (
 	"github.com/NebulousLabs/Sia/modules"
 )
 
-// A Download is a download that has been queued by the renter.
+var (
+	downloadAttempts = 5
+)
+
+// A Download is a file download that has been queued by the renter.
 type Download struct {
-	completed   bool
+	complete    bool
+	filesize    uint64
+	received    uint64
 	destination string
 	nickname    string
 
-	renter *Renter
+	pieces  []FilePiece
+	file    *os.File
+	gateway modules.Gateway
```

Comment:
I actually quite prefer the atomic implementation. The need for a mutex is eliminated because only one field is ever updated (`received`), and the update happens atomically.